Upstream commit: 68952421e6fb8a161faf40861d122f35a1e058fc
ChangeLog / commit message:
Correct test for fpr pairs.
Files: gas/config/tc-mips.c

--- a/gas/config/tc-mips.c
+++ b/gas/config/tc-mips.c
@@ -1191,11 +1191,11 @@
 	 because there is no instruction that sets both $f0 and $f1
 	 and requires a delay.  */
       if ((ip->insn_mo->pinfo & INSN_READ_FPR_S)
-	  && ((((ip->insn_opcode >> OP_SH_FS) & OP_MASK_FS) &~(unsigned)1)
+	  && (((ip->insn_opcode >> OP_SH_FS) & OP_MASK_FS)
 	      == (reg &~ (unsigned) 1)))
 	return 1;
       if ((ip->insn_mo->pinfo & INSN_READ_FPR_T)
-	  && ((((ip->insn_opcode >> OP_SH_FT) & OP_MASK_FT) &~(unsigned)1)
+	  && (((ip->insn_opcode >> OP_SH_FT) & OP_MASK_FT)
 	      == (reg &~ (unsigned) 1)))
 	return 1;
     }
@@ -1286,7 +1286,7 @@
 
       for (l = insn_labels; l != NULL; l = l->next)
 	{
-#ifdef OBJ_ELF
+#ifdef S_SET_OTHER
 	  if (OUTPUT_FLAVOR == bfd_target_elf_flavour)
 	    S_SET_OTHER (l->label, STO_MIPS16);
 #endif
@@ -8664,7 +8664,7 @@
 
   /* If we aren't adjusting this fixup to be against the section
      symbol, we need to adjust the value.  */
-#ifdef OBJ_ELF
+#ifdef S_GET_OTHER
   if (fixP->fx_addsy != NULL
       && OUTPUT_FLAVOR == bfd_target_elf_flavour
       && S_GET_OTHER (fixP->fx_addsy) == STO_MIPS16)
@@ -10053,7 +10053,7 @@
     return 0;
   if (fixp->fx_addsy == NULL)
     return 1;
-#ifdef OBJ_ELF
+#ifdef S_GET_OTHER
   if (OUTPUT_FLAVOR == bfd_target_elf_flavour
       && S_GET_OTHER (fixp->fx_addsy) == STO_MIPS16
       && fixp->fx_subsy == NULL)
